### PR TITLE
[Catalog Promotions][Docs] Add a section about how promotions are applied

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: docs/book/products/catalog_promotions.rst [hunk 1]
```diff
@@ -273,6 +275,28 @@ Catalog Promotion Actions configuration reference
 | ``percentage_discount``       | ``['amount' => $amountFloat]``                                     |
 +-------------------------------+--------------------------------------------------------------------+
 
+How the Catalog Promotions are applied?
+---------------------------------------
+
+The Catalog Promotion application process utilises `Doctrine events <https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html>`_.
+When the new Promotion is created there is an event ``postPersist()`` in `CatalogPromotionEventListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php>`_
```
Comment (CoderMaggie) [inline docs/book/products/catalog_promotions.rst:282]:
```suggestion
When a new Promotion is created there is an event ``postPersist()`` in `CatalogPromotionEventListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php>`_
```

File: docs/book/products/catalog_promotions.rst [hunk 2]
```diff
@@ -273,6 +275,28 @@ Catalog Promotion Actions configuration reference
...
+that dispatches another ``CatalogPromotionUpdated`` event to event bus.
+
+This event is handled by `CatalogPromotionUpdateListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Listener/CatalogPromotionUpdateListener.php>`_ which resolves appropriate ``CatalogPromotion``.
```
Comment (CoderMaggie) [inline docs/book/products/catalog_promotions.rst:285]:
```suggestion
This event is handled by `CatalogPromotionUpdateListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Listener/CatalogPromotionUpdateListener.php>`_ which resolves the appropriate ``CatalogPromotion``.
```

File: docs/book/products/catalog_promotions.rst [hunk 3]
```diff
@@ -273,6 +275,28 @@ Catalog Promotion Actions configuration reference
...
+which inside uses the `CatalogPromotionApplicator <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/Applicator/CatalogPromotionApplicator.php>`_.
+
+The **CatalogPromotionProcessor**'s method ``process()`` is executed on the eligible items:
+
+* firstly it iterates over eligible items: ``Product Variants``,
+* then it calculates and applies the ``CatalogPromotionAction`` for given item
+
+.. note::
+
+    If you want to reapply Catalog Promotion manually you can refer to :ref:`How to create a Catalog Promotion Rule and Action? <how-to-create-a-catalog-promotion-rule-and-action>` section
```
Comment (CoderMaggie) [inline docs/book/products/catalog_promotions.rst:298]:
```suggestion
    If you want to reapply Catalog Promotion manually you can refer to the :ref:`How to create a Catalog Promotion Rule and Action? <how-to-create-a-catalog-promotion-rule-and-action>` section.
```

**Review by AdamKasp — APPROVED** [review]

File: docs/book/products/catalog_promotions.rst [hunk 4]
```diff
@@ -273,6 +275,28 @@ Catalog Promotion Actions configuration reference
 | ``percentage_discount``       | ``['amount' => $amountFloat]``                                     |
 +-------------------------------+--------------------------------------------------------------------+
 
+How the Catalog Promotions are applied?
+---------------------------------------
+
+The Catalog Promotion application process utilises `Doctrine events <https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html>`_.
+When a new Promotion is created there is an event ``postPersist()`` in `CatalogPromotionEventListener <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php>`_
```
Comment (Zales0123) [inline docs/book/products/catalog_promotions.rst:282]: In fact `postPersist()` is a **function** not an **event** (a function that is triggered on `postPersist` event 💃)

**Comment by Zales0123** [comment]
Thanks, @arti0090! :1st_place_medal:

**Pull request merged**